### ISSUE TITLE
[Research] How to handle click events between devices

### ISSUE THREAD

**Issue by ilmoeuro** [otavanopisto/muikku#3093]
#### Research conclusion

During the research for the solution of the issue with the click events, several problems were identified, being some:

Misbehaviour of the jquery draggable interactions which caused scrolling to fail.
Misbehaviour of the global event listeners in newer versions of desktop safari.

Several libraries were identified, being actually way too many out there; however the best contendant was https://github.com/jquery/PEP a polyfill for the native pointer events.

However the main problem with all these approaches is that they all require heavy refactoring of the code, which is anyway being done already during the redesign; also sometimes is better to keep behaviours separated, depending to what wants to be done; also given the browsers we are targeting most likely the touch events will suffice, the inherent problem was that this webpage was designed without mobile in mind to start with.

My conclusion for the long term (and the refactor process) is to have a way to test that everything works just right and hotfix things that don’t for the current muikku version.

For current fixes such as the jquery draggable for example, let’s make a hotfix where the dragging function is overridden by an user defined function that doesn’t do anything in mobile devices only.

And it seems that the global issue is because the listeners are appended somehow when the element isn’t ready or because the element doesn't listen to click events being the document itself, a replacement to body may fix the issue.

We should follow a process of testing with refactoring as we apply the new design provided by lankinen; using the devices that are representative to the target publics we approach, currently we have no problem with general firefox and chrome checking, as well as safari for desktop, but we’d need the following to test:

 - Safari + Chrome iphone or ipad.
 - Windows mobile.
 - Internet explorer 10.

as long as we can prove it works properly in all the platforms there may not be a need for using external libraries to handle events, maybe only on the fast click issues as referred to this article http://developer.telerik.com/featured/300-ms-click-delay-ios-8/ we can use some of their solutions for each platform, where chrome and firefox don’t need any, internet explorer is about css and ios8 and 7 are the main problem.

So there is no simple fix because there’s no centralized way in which events were written, and my recommendation is to stay away from refactoring code twice; and just follow the redesign process to fix the issues found on the way as we check the design with the devices, while shipping hotfixes for issues that users complain about.

So in conclusion:
 - <strike>Fast fix the dragging/scrolling issue.</strike>
 - Obtain the devices for testing and assemble a way to test the design.
 - Wait for Lankinen for his next designs.
 - Keep developing on the new design.
